### ISSUE TITLE
yolov8 Removes the small target detection layer

### ISSUE THREAD

**Issue by BILIBILINB** [ultralytics/ultralytics#9317]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello Sir, I would like to ask yolov8, after removing the small target detection layer, the size of his model has increased and the parameters used have also increased, what is the reason?
This is the model after MobileNetV3
                   from  n    params  module                                       arguments
  0                  -1  1       232  ultralytics.nn.modules.conv.Conv_BN_HSwish   [3, 8, 2]
  1                  -1  1       628  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[8, 8, 16, 3, 2, 1, 0]

  2                  -1  1      2696  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[8, 16, 72, 3, 2, 0, 0]

  3                  -1  1      3992  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[16, 16, 88, 3, 1, 0, 0]

  4                  -1  1     11400  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[16, 24, 96, 5, 2, 1, 1]

  5                  -1  1     47628  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[24, 24, 240, 5, 1, 1, 1]

  6                  -1  1     47628  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[24, 24, 240, 5, 1, 1, 1]

  7                  -1  1     16638  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[24, 24, 120, 5, 1, 1, 1]

  8                  -1  1     21684  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[24, 24, 144, 5, 1, 1, 1]

  9                  -1  1     71016  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[24, 48, 288, 5, 2, 1, 1]

 10                  -1  1    238704  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[48, 48, 576, 5, 1, 1, 1]

 11                  -1  1    238704  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[48, 48, 576, 5, 1, 1, 1]

 12                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 13             [-1, 8]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 14                  -1  1    108288  ultralytics.nn.modules.block.C2f             [72, 128, 1]
 15                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 16             [-1, 3]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 17                  -1  1     34176  ultralytics.nn.modules.block.C2f             [144, 64, 1]
 18                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]
 19            [-1, 14]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 20                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]
 21                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]
 22            [-1, 11]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 23                  -1  1    439808  ultralytics.nn.modules.block.C2f             [176, 256, 1]
 24        [17, 20, 23]  1    751507  ultralytics.nn.modules.head.Detect           [1, [64, 128, 256]]
YOLOv87 summary: 307 layers, 2343081 parameters, 2343065 gradients, 5.8 GFLOPs

This is the result after removing the small target detection on top of the above
                  from  n    params  module                                       arguments
  0                  -1  1       232  ultralytics.nn.modules.conv.Conv_BN_HSwish   [3, 8, 2]
  1                  -1  1       628  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[8, 8, 16, 3, 2, 1, 0]

  2                  -1  1      2696  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[8, 16, 72, 3, 2, 0, 0]

  3                  -1  1      3992  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[16, 16, 88, 3, 1, 0, 0]

  4                  -1  1     11400  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[16, 24, 96, 5, 2, 1, 1]

  5                  -1  1     47628  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[24, 24, 240, 5, 1, 1, 1]

  6                  -1  1     47628  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[24, 24, 240, 5, 1, 1, 1]

  7                  -1  1     16638  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[24, 24, 120, 5, 1, 1, 1]

  8                  -1  1     21684  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[24, 24, 144, 5, 1, 1, 1]

  9                  -1  1     71016  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[24, 48, 288, 5, 2, 1, 1]

 10                  -1  1    238704  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[48, 48, 576, 5, 1, 1, 1]

 11                  -1  1    238704  ultralytics.nn.modules.conv.MobileNetV3_InvertedResidual[48, 48, 576, 5, 1, 1, 1]

 12                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 13             [-1, 8]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 14                  -1  1    108288  ultralytics.nn.modules.block.C2f             [72, 128, 1]
 15                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]
 16            [-1, 11]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 17                  -1  1    439808  ultralytics.nn.modules.block.C2f             [176, 256, 1]
 18            [14, 17]  1   1042322  ultralytics.nn.modules.head.Detect           [1, [128, 256]]
YOLOv871 summary: 255 layers, 2439080 parameters, 2439064 gradients, 3.6 GFLOPs

**Comment by glenn-jocher** [comment]
@BILIBILINB hello! It's great to see your in-depth observation with YOLOv8 after removing the small target detection layer. 🕵️

The increase in model size and parameters, despite removing a layer, could be due to the architectural changes that follow layer removal. Removing layers from a deep neural network like YOLOv8, especially those designed for specific target sizes, alters the flow and processing of feature maps. This may necessitate compensatory adjustments in the network to maintain performance, leading to an increase in parameters in subsequent layers or additional computation elsewhere in the model. Essentially, the network tries to adapt to these changes, which can inadvertently increase complexity.

It's also possible that the report reflects the total parameters inclusive of some form of additional structural or connective adaptations the model undergoes post-removal, ensuring that the remaining architecture still functions optimally for its detection tasks.

If you're looking into optimizing the model further, consider reviewing how each layer and its connections contribute to the model's final size and performance, and experiment with different configurations to find a balance that suits your needs.

Hope this helps! Let us know if you dive deeper or have more questions. 🚀

**Comment by BILIBILINB** [comment]
Thank you for your answer to my confusion.I will continue to try new methods for this research.Wish you a happy life.

**Comment by glenn-jocher** [comment]
@BILIBILINB You're welcome! 😊 I'm glad I could help clarify things for you. It sounds like you're on an exciting path with your research, and I'm sure you'll discover many insightful methods along the way. If you ever need more guidance or want to share your progress, feel free to reach out. Wishing you all the best in your endeavors, and may you have a joyful and fulfilling journey ahead! 🚀
